### ISSUE TITLE
Inconsistent column name in the generated SQL

### ISSUE THREAD

**Issue by mngr** [metabase/metabase#39709]
### Describe the bug

When you use query builder and summarize by any column like Payment for example it automatically creates a column with a name like Sum of Payment.
But in the generated SQL it is named just as "sum" so the name of the column doesn't match what is set in the query builder
<img width="1028" alt="Captura de ecrã 2024-03-06, às 17 36 31" src="https://github.com/metabase/metabase/assets/777800/47d3f810-a66b-4265-86a0-bd9614756b19">
<img width="595" alt="Captura de ecrã 2024-03-06, às 17 36 48" src="https://github.com/metabase/metabase/assets/777800/132deb7d-1837-4e7a-a916-e1407c5f6382">
<img width="553" alt="Captura de ecrã 2024-03-06, às 17 37 12" src="https://github.com/metabase/metabase/assets/777800/598bcf11-365e-495a-b98c-2744b96559a9">

### To Reproduce

1. Go to the query builder
2. Select Sample Database -> Invoices in the data picker
3. Click on Summarize
4. Add Sum of... for Payment column
5. Hit Preview, you'll see a column Sum of Payment created
6. Open SQL generated, you'll see there `SUM("PUBLIC"."INVOICES"."PAYMENT") AS "sum"`

### Information about your Metabase installation

```JSON
last version
```

### Severity

annoying

**Comment by ranquild** [comment]
I don't think we should change the name of the column in SQL if it's renamed in the UI.
